Design.md

```diff
@@ -309,3 +332,29 @@ Get the platform usage of global (without query parameter), tenant (with tenant_
         * `key` : `offset` and `value` : `marker`
     1. `AssumeRoleCache` for all the `User` and `S3 Credential` APIs
         * `key` : `RoleArn` and `value` : `Credentials`
+
+## SecretKey Encryption Strategy
+
+1. For encrypting/ decrypting the `secret-keys` a `cipher` algorithm, and a `secret-password` will be used. These properties will be provided in `crypto.properties` config file.
+1. A `key-id` property will be provided for each `secret-password` in the `crypto.properties` config file to maintain the version of the key and will be used in the Key rotation.
+1. Meta-information will be encoded with base64 and added at the beginning of the encrypted string. This meta information will include the `key-id` of the `secret-password` used.
```

Comment:
This will also need to include the nonce used, most likely?

Reply:
Fixed in 380d8acd7a40b01860989f28d0eb985b95fc5a61